daliuge-engine/dlg/apps/pyfunc.py

```diff
@@ -498,9 +502,17 @@ def initialize_with_func_code(self):
         """
         This function takes over if code is passed in through an argument.
         """
+        serialized = False
         if not isinstance(self.func_code, bytes):
-            self.func_code = base64.b64decode(self.func_code.encode("utf8"))
-        self.f = import_using_code(self.func_code)
+            try:
+                self.f = import_using_code(self.func_code, serialized=False)
```

Comment:
Given we have the `serialized` variable above: 


```suggestion
                self.f = import_using_code(self.func_code, serialized=serialized)
```